runtime/src/migrations/erc20_peg.rs

```diff
@@ -211,5 +241,61 @@ pub mod v1 {
 				);
 			});
 		}
+
+		#[test]
+		fn migration_test_k() {
+			new_test_ext().execute_with(|| {
+				StorageVersion::new(0).put::<Erc20Peg>();
+
+				let one: BlockNumberFor<Runtime> = 1;
+				let two: BlockNumberFor<Runtime> = 2;
+
+				let block_key_1 = Twox64Concat::hash(&one.encode());
+				let block_key_2 = Twox64Concat::hash(&two.encode());
+
+				let delayed_payment_1: OldDelayedPaymentSchedule =
+					BoundedVec::truncate_from(vec![1, 2, 3, 4, 5]);
+				let delayed_payment_2: OldDelayedPaymentSchedule =
+					BoundedVec::truncate_from(vec![6, 7, 8, 9, 10]);
+
+				Map::unsafe_storage_put::<OldDelayedPaymentSchedule>(
+					b"Erc20Peg",
+					b"DelayedPaymentSchedule",
+					&block_key_1,
+					delayed_payment_1.clone(),
+				);
+				Map::unsafe_storage_put::<OldDelayedPaymentSchedule>(
+					b"Erc20Peg",
+					b"DelayedPaymentSchedule",
+					&block_key_2,
+					delayed_payment_2.clone(),
+				);
+
+				Upgrade::on_runtime_upgrade();
+				assert_eq!(Erc20Peg::on_chain_storage_version(), 1);
+
+				let expected_delayed_payment_1: NewDelayedPaymentSchedule =
+					WeakBoundedVec::force_from(delayed_payment_1.into_inner(), None);
+				let expected_delayed_payment_2: NewDelayedPaymentSchedule =
+					WeakBoundedVec::force_from(delayed_payment_2.into_inner(), None);
+
+				assert_eq!(
+					Map::unsafe_storage_get::<NewDelayedPaymentSchedule>(
+						b"Erc20Peg",
+						b"DelayedPaymentSchedule",
+						&block_key_1
+					),
+					Some(expected_delayed_payment_1)
+				);
+				assert_eq!(
+					Map::unsafe_storage_get::<NewDelayedPaymentSchedule>(
+						b"Erc20Peg",
+						b"DelayedPaymentSchedule",
+						&block_key_2
+					),
+					Some(expected_delayed_payment_2)
+				);
```

Comment:
At the end of the test, the StorageMap should be able to decode the type properly using the standard get method, so you can use standard gets here
```suggestion
				assert_eq!(
					DelayedPaymentSchedule::<Runtime>::get(one),
					expected_delayed_payment_1
				);
				assert_eq!(
					DelayedPaymentSchedule::<Runtime>::get(two),
					expected_delayed_payment_2
				);
```